src/Drivers/OutputGPIO.cpp

```diff
@@ -0,0 +1,27 @@
+#include "OutputGPIO.h"
+#include "Log/Log.h"
+
+OutputGPIO::OutputGPIO(const std::vector<OutputPinDef>& outputs, StrongObjectPtr<GPIO> gpio) : OutputDriver(outputs), gpio(std::move(gpio)){
+	for(const auto& output: this->outputs){
+		const auto pin = (gpio_num_t) output.port;
+		gpio->setMode(pin, GPIOMode::Output);
+
+	}
+}
+
+void OutputGPIO::write(int port, bool value) noexcept{
+	Super::write(port, value);
+
+	const auto pin = (gpio_num_t) port;
+	gpio->write(pin, value ^ inversions[port]);
```

Comment:
Svaki put kad na mapi koristis operator [] bez da prije provjeris jel taj key vec postoji unutra, dodas u mapu daj key sa default value sto je best case scenario waste of memory and performance ali u nekim slucajevima mozes bez da zelis dodat npr port koji ce se skenirati za input a nije intended for it, ili port na koji se moze writeat a nije intended for it